lib/index.ts

```diff
@@ -67,14 +68,22 @@ class PWMetrics {
 
     for (let runIndex of runs) {
       try {
-        metricsResults[runIndex] = await this.run();
+        const currentMetricResult: MetricsResults = await this.run();
+        currentMetricResult.error = this.resultHasExpectationErrors(currentMetricResult);
+        metricsResults[runIndex] = currentMetricResult;
         console.log(messages.getMessageWithPrefix('SUCCESS', 'SUCCESS_RUN', runIndex, runs.length));
       } catch (error) {
         metricsResults[runIndex] = error;
         console.error(messages.getMessageWithPrefix('ERROR', 'FAILED_RUN', runIndex, runs.length, error.message));
       }
     }
 
+    const hasMetricResultsError = metricsResults.filter(m => !m.error).length > 0;
+
+    if (hasMetricResultsError && this.flags.expectations) {
+      throw new Error(messages.getMessage('HAS_EXPECTATION_ERRORS'));
```

Comment:
I don't agree with this approach. Just because a particular metric failed, it does not mean that the rest of the metrics should not be shown. You should return the output as is currently implemented (both the graphics and json output) with an appropriate error process code if any expectation was not meet.